CMSgov/managed-care-review
# State user sees submissions with unlocked status on state dashboard

## Summary
As a state user, when I am on my state dashboard I see if a previously submitted submission package has been unlocked, so that I know which submissions still need to be edited and resubmitted. 

DRAFT PR BECAUSE I AM FIXING UP WITH BOTH #827 and main so the diff is a little off until that merges

#### Related issues
https://qmacbis.atlassian.net/browse/OY2-13844

#### Screenshots

#### Test cases covered
[api tests] `indexSubmission2` query
- returns a list of submissions that includes newly created entries
- synthesizes the right statuses as a submission is submitted/unlocked/etc
- a different user from the same state can index submissions'
- returns no submissions for a different state's user

[web tests] dashboard page
- adjusted table test to include an unlocked submission

## QA guidance
- open up the app in two tabs (on cognito). log into app as cms user on one side
- create submission, submit submission as a state user
- unlock submission as a cms user
- submission page appears unlocked as cms user
- go to edit unlocked submission as state user (should show unlocked tag in dashboard and navigate to review and submit page)
- submit unlocked submission again as state user
- go to dashboard page as state user, the just resubmitted submissions  should have the same treatment as a submitted submission 
- cms user should see the submission they just unlocked as locked
